crates/exec-wasmtime/src/runtime/io/mod.rs

```diff
@@ -4,15 +4,20 @@
 
 pub mod null;
 
-use wasi_common::file::FileCaps;
+use wasi_common::file::FileAccessMode;
 use wasi_common::WasiFile;
 
-pub fn stdio_file(mut file: impl WasiFile + 'static) -> (Box<dyn WasiFile>, FileCaps) {
+// This import provides the trait needed for `FileAccessMode::all()`, yet Clippy
+// claims the import isn't used. Remove this import and the code fails to compile!
+#[allow(unused_imports)]
+use wiggle::bitflags::Flags;
+
+pub fn stdio_file(mut file: impl WasiFile + 'static) -> (Box<dyn WasiFile>, FileAccessMode) {
     // Ensure wasmtime can detect the TTY.
-    let caps = if file.isatty() {
-        FileCaps::all().difference(FileCaps::TELL | FileCaps::SEEK)
+    let mode = if file.isatty() {
+        FileAccessMode::all()
     } else {
-        FileCaps::all()
+        FileAccessMode::READ
```

Comment:
Why not `all()`? I thought the main purpose was to take away TELL and SEEK in case of a tty... of course that does not cover the case of a pipe.

Reply:
stdio_file(stdout) only returning `FileAccessMode::READ` makes no sense somehow

Reply:
If there is no `FileCaps` anymore you can maybe remove this function altogether.

Reply:
I wasn't sure since I don't have the context as to why some decisions were made, or what was intended to be accomplished.

Reply:
It was done to mark the capabilities of the file descriptor... so a pipe does not have `tell` and `seek` capabilities.

FileAccessMode might not make sense here.